src/full/Agda/TypeChecking/Quote.hs

```diff
@@ -185,14 +185,13 @@ quotingKit = do
             in normalClause !@ quotePats ps @@ quoteTerm v
 
       list :: [ReduceM Term] -> ReduceM Term
-      list []       = pure nil
-      list (a : as) = cons !@ a @@ list as
+      list = foldr (\ a -> (@@) (cons !@ a)) (pure nil)
```

Comment:
```suggestion
      list = foldr (\ a as -> cons !@ a @@ as) (pure nil)
```
Infix applications are not more readable eta contracted.

Reply:
Done.